src/tests/lvm/common/util_resize.go

```diff
@@ -0,0 +1,156 @@
+		// after fio completes sleep of a long time
+		PostOpSleep: 600000,
+	}
+
+	loopDevice := e2e_agent.LoopDevice{
+		Size:   10737418240,
+		ImgDir: "/tmp",
+	}
+
+	workerNodes, err := lvm.ListLvmNode(common.NSOpenEBS())
+	Expect(err).ToNot(HaveOccurred(), "failed to list worker node")
+
+	NodeConfig = lvm.LvmNodesDevicePvVgConfig{
+		VgName:        "lvmvg",
+		NodeDeviceMap: make(map[string]e2e_agent.LoopDevice), // Properly initialize the map
+	}
+	for _, node := range workerNodes {
+		NodeConfig.NodeDeviceMap[node] = loopDevice
+	}
+
+	logf.Log.Info("setup node with loop device, pv and vg", "node config", NodeConfig)
+	err = NodeConfig.ConfigureLvmNodesWithDeviceAndVg()
+	Expect(err).ToNot(HaveOccurred(), "failed to setup node")
+
+	// setup sc parameters
+	app.Lvm = k8stest.LvmOptions{
+		VolGroup:      NodeConfig.VgName,
+		Storage:       "lvm",
+		ThinProvision: thinProvisioned,
+	}
+
+	logf.Log.Info("create sc, pvc, fio pod")
+	err = app.DeployApplication()
+	Expect(err).To(BeNil(), "failed to deploy app")
+
+	// sleep for 30 seconds before resizing volume
+	logf.Log.Info("Sleep before resizing volume", "duration", volume_resize.DefSleepTime)
+	time.Sleep(time.Duration(volume_resize.DefSleepTime) * time.Second)
+
+	expandedVolumeSizeMb := app.VolSizeMb + 1024
+	// expand volume by editing pvc size
+	logf.Log.Info("Update volume size", "new size in MiB", expandedVolumeSizeMb)
+	_, err = k8stest.UpdatePvcSize(app.GetPvcName(), common.NSDefault, expandedVolumeSizeMb)
+	Expect(err).ToNot(HaveOccurred(), "failed to expand volume %s, error: %v", app.GetPvcName(), err)
+
+	// verify pvc capacity to new size
+	logf.Log.Info("Verify pvc resize status")
+	pvcResizeStatus, err := volume_resize.WaitForPvcResize(app.GetPvcName(), common.NSDefault, expandedVolumeSizeMb)
+	Expect(err).ToNot(HaveOccurred(), "failed to verify resized pvc %s, error: %v", app.GetPvcName(), err)
+	Expect(pvcResizeStatus).To(BeTrue(), "failed to resized pvc %s, error: %v", app.GetPvcName(), err)
+
+	// Check fio pod status
+	logf.Log.Info("Check fio pod status")
+	phase, _, err := k8stest.CheckFioPodCompleted(app.GetPodName(), common.NSDefault)
+	Expect(err).To(BeNil(), "CheckPodComplete got error %s", err)
+	Expect(phase).ShouldNot(Equal(coreV1.PodFailed), "fio pod phase is %s", phase)
+
+	// wait for fio completion - monitoring log output
+	exitValue, fErr := app.WaitFioComplete(volume_resize.DefFioCompletionTime, 5)
+	Expect(fErr).ToNot(HaveOccurred())
+	logf.Log.Info("fio complete", "exit value", exitValue)
+	Expect(exitValue == 0).Should(BeTrue(), "fio exit value is not 0")
+
+	// print fio target sizes retrieved by monitoring log output
+	ftSizes, ffErr := app.FioTargetSizes()
+	Expect(ffErr).ToNot(HaveOccurred())
+	for path, size := range ftSizes {
+		logf.Log.Info("ftSize (poc_resize_1)", "path", path, "size", volume_resize.ByteSizeString(size), "bytes", size)
+		ftSize1 = size
+	}
+	Expect(len(ftSizes)).To(BeNumerically("==", 1), "unexpected fio target sizes")
+
```

Comment:
Just to be on the safe side - do `app.WaitComplete` before firing up the 2nd application instance.

Reply:
it is already there at 101